sandy/fy.py

```diff
@@ -203,6 +213,140 @@ def _expand_zam(self):
                            dtype=int)
         return self.data.assign(Z=zam.Z, A=zam.A, M=zam.M)
 
+    def custom_perturbation(self, zam, mt, e, zap, pert):
+        """
+        Apply a custom perturbation in the fission yields identified by the mt,
+        for a given zam in a given energy and for a given product.
+
+        Parameters
+        ----------
+        zam : `int`
+            ZAM number of the material to which perturbations are to be
+            applied.
+        mt : `int`
+            MT reaction number of the FY to which perturbations are to be
+            applied either 454 or 459.
+        e : `float`
+            Energy of the fissioning system to which which perturbations
+            are to be applied.
+        zap : `int`
+            ZAP of the product to which perturbations are to be
+            applied.
+        pert : `float`
+            Perturbation coefficients as ratio values.
+
+        Returns
+        -------
+        `Fy`
+            Fission yield instance with applied perturbation.
+
+        Examples
+        --------
+        >>> tape = sandy.get_endf6_file("jeff_33", 'nfpy', 'all')
+        >>> nfpy = Fy.from_endf6(tape)
+        >>> nfpy_pert = nfpy.custom_perturbation(922350, 459, 0.0253, 551370, -0.1)
+        >>> comp = nfpy_pert.data.query('ZAM==922350 & ZAP==551370 & MT==459 & E==0.0253').squeeze().FY
+        >>> assert np.setdiff1d(nfpy_pert.data.values, nfpy.data.values) == comp
+        """
+        df = self.data.copy()
+        mask = (df.ZAM == zam) & (df.ZAP == zap) & (df.MT == mt) & (df.E == e)
+        df.loc[mask, "FY"] = df[mask].squeeze().FY * (1 + pert)
+        return self.__class__(df)
+
+    def apply_bmatrix(self, zam, e, decay_data):
+        """
+        Perform IFY = (1-B)CFY equation to calculate IFY in a given zam
+        for a given energy and apply into the original data.
+
+        Parameters
+        ----------
+        zam : `int`
+            ZAM number of the material to which calculations are to be
+            applied.
+        e : `float`
+            Energy to which calculations are to be applied.
+        decay_data : `sandy.DecayData`
+            Radioactive nuclide data for several isotopes.
+
+        Returns
+        -------
+        `Fy`
+            Fission yield instance with IFY calculated for a given combination
+            of ZAM/e/decay_data.
+
+        Examples
+        --------
+        >>> zam = [591480, 591481, 601480]
+        >>> decay_minimal = sandy.get_endf6_file("jeff_33", 'decay', zam)
+        >>> decay_fytest = sandy.DecayData.from_endf6(decay_minimal)
+        >>> npfy = Fy(minimal_fytest_2)
+        >>> npfy_pert = npfy.apply_bmatrix(942390, 5.00000e+05, decay_fytest)
```

Comment:
I still believe you cannot use the B-matrix this way....I mean, in this test

Reply:
The aim of the test is: 
Select related cores by decay (so that the B matrix is not zero) and create a variable (minimal_fytest_2) with these cores and invented IFY and CFY values. 
As IFY and CFY are invented, when applying bmatrix, all IFY values will change, since IFY and CFY are not related. In this way, by manually performing the multiplication between the matrix (1-B) and the vector CFY, we can check that the method results are correct.  
Finally, we compare the values of the original FY object and the one to which we have applied bmatrix. The only different values have to be the IFY values.